unittests/test_check_filters.py

```diff
@@ -0,0 +1,93 @@
+import unittest
+
+import reframe.core.runtime as rt
+import reframe.frontend.check_filters as filters
+import reframe.utility.sanity as sn
+import unittests.fixtures as fixtures
+from reframe.core.pipeline import RegressionTest
+
+
+class TestCheckFilters(unittest.TestCase):
+    def create_check(self, attrs):
+        ret = RegressionTest()
+        for k, v in attrs.items():
+            setattr(ret, k, v)
+
+        return ret
+
+    def setUp(self):
+        self.checks = [
+            self.create_check({
+                'name': 'check1',
+                'tags': {'a', 'b', 'c', 'd'},
+                'valid_prog_environs': ['env1', 'env2'],
+                'valid_systems': ['testsys:gpu', 'testsys2:mc'],
+                'num_gpus_per_node': 1}),
+            self.create_check({
+                'name': 'check2',
+                'tags': {'x', 'y', 'z'},
+                'valid_prog_environs': ['env3'],
+                'valid_systems': ['testsys:mc', 'testsys2:mc'],
+                'num_gpus_per_node': 0}),
+            self.create_check({
+                'name': 'check3',
+                'tags': {'a', 'z'},
+                'valid_prog_environs': ['env3', 'env4'],
+                'valid_systems': ['testsys:gpu'],
+                'num_gpus_per_node': 1})
+        ]
+
+    def test_have_name(self):
+        self.assertEqual(1, sn.count(filter(filters.have_name('check1'),
+                                            self.checks)))
+        self.assertEqual(1, sn.count(filter(filters.have_name('check2'),
+                                            self.checks)))
+        self.assertEqual(1, sn.count(filter(filters.have_name('check3'),
+                                            self.checks)))
+        self.assertEqual(0, sn.count(filter(filters.have_name('check4'),
+                                            self.checks)))
+
+    def test_have_not_name(self):
+        self.assertEqual(1, sn.count(filter(filters.have_not_name('check1'),
+                                            self.checks)))
+        self.assertEqual(1, sn.count(filter(filters.have_not_name('check2'),
```

Comment:
Shouldn't this and the next one be equal to `2`?

Reply:
Yes, I haven't run the unit tests again. I am fixing it now and will re-push.